SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialPackageMapClient.cpp

```diff
@@ -269,6 +270,50 @@ void USpatialPackageMapClient::ClearRemovedDynamicSubobjectObjectRefs(const Work
 	}
 }
 
+void USpatialPackageMapClient::RemoveBNetLoadOnClientRuntimeRemovedComponents(const Worker_EntityId& EntityId,
+																			  const TArray<SpatialGDK::ComponentData>& NewComponents,
+																			  const SpatialGDK::ActorSystem& ActorSystem
+
+)
+{
+	auto ContainedInComponentsArr = [this, &NewComponents, &EntityId](const FUnrealObjectRef CheckComponentObjRef) {
+		for (const SpatialGDK::ComponentData& Component : NewComponents)
+		{
+			uint32 Offset = 0;
+			NetDriver->ClassInfoManager->GetOffsetByComponentId(Component.GetComponentId(), Offset);
+
+			if (FUnrealObjectRef(EntityId, Offset) == CheckComponentObjRef)
+			{
+				return true;
+			}
+		}
+		return false;
+	};
+
+	// Go over each stored sub-object and determine whether it is contained within the new components array
+	// If it is not contained within the new components array, it means the sub-object was removed while out of the client's interest
+	// If so, remove it now
+	for (auto DynamicSubObjectIterator = RemovedDynamicSubobjectObjectRefs.CreateIterator(); DynamicSubObjectIterator;
+		 ++DynamicSubObjectIterator)
+	{
+		if (DynamicSubObjectIterator->Key.Entity == EntityId)
+		{
+			if (!ContainedInComponentsArr(DynamicSubObjectIterator->Key))
```

Comment:
hmhmhm, maybe there is an easier way to do this?
What if we could do a set difference like NewComponents \ RemovedDynamicComponents?
Not actually sure if it's feasible, but would make this simpler, maybe even faster.

Reply:
I think it can be done but have to construct both the sets first. I figure this overhead of stack allocation (and sets themselves) would be slower unless you had a v. large amount of components. Do you think it would be more readable with this approach?

Reply:
I was thinking more that we would add new persistent set structures, so we don't have to construct them here in-place.
But it is a fairly large change, would have to think about it a bit (or you can think about it a bit :D).

Reply:
Yeah, I guess it could make fixing the static sub-objects bug easier. Whether this is worth probably depends on whether this would simplify other code or not.

Reply:
(and how hard it would be to do static sub-objects as another case/combine it with this, which I'm not too sure about)